### PR TITLE
Add fast listing component to quick demo

### PR DESCRIPTION
**Pull request by bernardhan33**
Adds the fast listing demo to the quick walkthrough. It runs against a 2M public dataset hosted under our public directory. Since the default free Google Colab Runtime has only 2 CPUs, it can only speed up listing by 2x.

If the UI is hard for you to review, please visit the [copy](https://colab.research.google.com/drive/1-yk9MiUnZ8HxRrvHmjxg3jRrg9rFDwPC?resourcekey=0-n5LguRQZD_3yA4TpwZbvHA#scrollTo=iRDqRYPuQflw) directly.

Feel free to leave any comments here or directly on the copy.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb [hunk 1]
```diff
@@ -647,6 +642,130 @@
     {
       "cell_type": "markdown",
       "source": [
+        "### Data Listing\n",
+        "\n",
+        "When working with cloud storage solutions such as GCS, data listing performance is critical because we will need it to support the map-style datasets' random access patterns.\n",
```
Comment (jdnurme) [inline demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb:647]: "When working with cloud storage solutions such as GCS, data listing performance is critical to enabling support for map-style datasets' random access patterns.

File: demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb [hunk 2]
```diff
@@ -647,6 +642,130 @@
...
+          },
+          "metadata": {},
+          "execution_count": 12
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "You can see that with 2 available CPUs from the Colab runtime, fast listing can speed up the listing time by 2x from 146 seconds to 69 seconds."
```
Comment (jdnurme) [inline demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb:759]: "Using the two available CPUs in the Colab runtime, fast listing can improve performance by 2x (from 146 to 69 seconds)."

**Review by jdnurme — APPROVED** [review]

File: demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb (hunk 1 above)
Comment (bernardhan33) [reply demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb]: done.

File: demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb (hunk 2 above)
Comment (bernardhan33) [reply demo/simple-walkthrough/Getting Started with Dataflux Dataset for PyTorch with Google Cloud Storage.ipynb]: done.

**Pull request merged**